### PR TITLE
Repeatable imports for ChassImporter

### PR DESCRIPTION
**Pull request by michellemtchai**
Instructions:
- close Docker
- run `docker-compose run rails-app rake db:drop db:create db:migrate db:seed db:seed:chass[round_110]`
  - This will give you the first mock_chass JSON to import

Testing:
- new command `docker-compose run rails-app rake db:seed:chass[file]`, where `file` is any JSON in the `db/seeds` folder
- run the following file to import sequentially to see the changes in the system data
  - `round_110_updated`
  - `round_110_add_apps`
  - `round_112`
- check UI views before running each new file to import to see the changes made

Changes:
- creates new `Application` for an `Applicant` if `Applicant` existed in a previous round
- updates `Applicant`, `Position`, `Application`, and `Instructor` if re-upload from the same round
- delete `Preference` object when applicant un-checks a course after the first upload
- delete `Position` to `Instructor` if instructor is removed from CHASS
- moved `app_id` to `Applicant` model
  - fixed rspec accordingly
  - fix `ChassExporter` accordingly

### CODE REVIEW

**Review by jmzaleski — APPROVED** [review]
wow, that's a bit PR.
The only bit that scared me was the destroy of the unused Preference.

File: app/services/chass_importer.rb [hunk 1]
```diff
@@ -31,23 +31,31 @@ def get_round_id
     end
   end
 
-  def insertion_helper(model, condition, exists)
-    unless condition
-      db_model = model
-      Rails.logger.debug db_model.attributes
+  def insertion_helper(model, data, ident, exists)
+    unless model.where(ident).exists?
+      db_model = model.create(data)
```
Comment (freeatnet) [inline app/services/chass_importer.rb:36]: We can use [`create!`](http://api.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-create-21) here to create the record and raise an exception.

File: app/services/chass_importer.rb [hunk 2]
```diff
@@ -31,23 +31,31 @@ def get_round_id
...
+      Rails.logger.debug "new #{JSON.pretty_generate(db_model.as_json)}\n\n"
       db_model.save!
+      return db_model
     else
       Rails.logger.debug "#{exists}"
+      db_model = model.find_by(ident)
+      Rails.logger.debug "existing model #{JSON.pretty_generate(db_model.as_json)}\n"
+      db_model.update_attributes(data)
```
Comment (freeatnet) [inline app/services/chass_importer.rb:44]: Same here, [`update_attributes!`](http://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-update-21) would work

File: app/services/chass_importer.rb (hunk 1 above)
Comment (michellemtchai) [reply app/services/chass_importer.rb:36]: @freeatnet Should I add this in a new PR?

Comment (freeatnet) [reply app/services/chass_importer.rb:36]: @michellemtchai I think this is something that can be put on the "when cleaning up" list :)

**Pull request merged**